packages/cid-verifier/src/verification.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-env serviceworker, browser */
+/* global Response */
+import pRetry from 'p-retry'
+import { getFromDenyList } from './utils/denylist'
+
+/**
+ * Get the google malware result stored in KV.
+ *
+ * @param {string} cid
+ * @param {import('./env').Env} env
+ */
+async function getGoogleMalwareResult (cid, env) {
+  const datastore = env.GOOGLE_EVALUATE_RESULTS
+  if (!datastore) {
+    throw new Error('db not ready')
+  }
+
+  return await pRetry(
+    () => datastore.get(cid),
+    { retries: 5 }
+  )
+}
+
+/**
+ * @param {Array<string>} params
+ * @param {(params: Array<string>, request: Request, env: import('./env').Env) => Promise<Response>} fn
+ * @returns {import('itty-router').RouteHandler<Request>}
+ */
+function withRequiredQueryParams (params, fn) {
+  /**
+   * @param {Request} request
+   * @param {import('./env').Env} env
+   */
+  return async function (request, env) {
+    const searchParams = (new URL(request.url)).searchParams
+
+    for (const param of params) {
+      if (!searchParams.get(param)) {
+        return new Response(`${param} is a required query param`, { status: 400 })
+      }
+    }
+
+    return await fn(params.map(param => String(searchParams.get(param))), request, env)
+  }
+}
+
+export const verificationGet = withRequiredQueryParams(['cid'],
+  /**
+   * Returns google malware result.
+   */
+  async function (params, request, env) {
+    const [cid] = params
+
+    const denyListResource = await getFromDenyList(cid, env)
+    if (denyListResource) {
+      const { status } = JSON.parse(denyListResource)
+      if (status === 451) {
+        return new Response('BLOCKED FOR LEGAL REASONS', { status: 451 })
+      } else {
+        return new Response('MALWARE DETECTED', { status: 403 })
+      }
+    }
+    return new Response('', { status: 204 })
+  }
+)
+
+/**
+ * Process CID with malware verification parties.
+ */
+export const verificationPost = withRequiredQueryParams(['cid', 'url'],
+  async function verificationPost (params, request, env) {
+    const [cid, url] = params
+    const googleMalwareResult = await getGoogleMalwareResult(cid, env)
+
+    if (!googleMalwareResult) {
+      env.GOOGLE_EVALUATE_RESULTS.put(cid, 'PENDING')
```

Comment:
Let's put everything within the same scope. So, move the put inside `fetchGoogleMalwareResults`. Also:
- this should have await
- this should have key as previously discussed: `cid/${google-service-id-name}`   https://hackmd.io/@vasco-santos/rJWZsz-Cc
- (thought we would go with the lock?)